hazelcast-client-new/src/main/java/com/hazelcast/client/config/ClientProperty.java

```diff
@@ -0,0 +1,143 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.instance.GroupProperty;
+import com.hazelcast.instance.HazelcastProperty;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.util.Preconditions.checkHasText;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Defines the name and default value for Hazelcast Client properties.
+ */
+public enum ClientProperty implements HazelcastProperty {
+
+    /**
+     * Client shuffles the given member list to prevent all clients to connect to the same node when
+     * this property is set to false. When it is set to true, the client tries to connect to the nodes
+     * in the given order.
+     */
+    SHUFFLE_MEMBER_LIST("hazelcast.client.shuffle.member.list", true),
+
+    /**
+     * Client sends heartbeat messages to the members and this is the timeout for this sending operations.
+     * If there is not any message passing between the client and member within the given time via this property
+     * in milliseconds, the connection will be closed.
+     */
+    HEARTBEAT_TIMEOUT("hazelcast.client.heartbeat.timeout", 60000, MILLISECONDS),
+
+    /**
+     * Time interval between the heartbeats sent by the client to the nodes.
+     */
+    HEARTBEAT_INTERVAL("hazelcast.client.heartbeat.interval", 5000, MILLISECONDS),
+
+    /**
+     * Number of the threads to handle the incoming event packets.
+     */
+    EVENT_THREAD_COUNT("hazelcast.client.event.thread.count", 5),
+
+    /**
+     * Capacity of the executor that handles the incoming event packets.
+     */
+    EVENT_QUEUE_CAPACITY("hazelcast.client.event.queue.capacity", 1000000),
+
+    /**
+     * Time to give up on invocation when a member in the member list is not reachable.
+     */
+    INVOCATION_TIMEOUT_SECONDS("hazelcast.client.invocation.timeout.seconds", 120, SECONDS);
+
+    private final String name;
+    private final String defaultValue;
+    private final TimeUnit timeUnit;
+
+    ClientProperty(String name, boolean defaultValue) {
+        this(name, defaultValue ? "true" : "false");
+    }
+
+    ClientProperty(String name, Integer defaultValue) {
+        this(name, String.valueOf(defaultValue));
+    }
+
+    ClientProperty(String name, String defaultValue) {
+        this(name, defaultValue, null);
+    }
+
+    ClientProperty(String name, Integer defaultValue, TimeUnit timeUnit) {
```

Comment:
@pveentjer Do you have any objections to merge this as an intermediate step? This is already done in this way on the member side, so right now it's inconsistent. Also we have to deprecate the old constants anyway, before you can replace it with a proper DI system.